src/rust/engine/graph/src/lib.rs

```diff
@@ -382,6 +382,42 @@ impl<N: Node> InnerGraph<N> {
       .filter_map(move |eid| self.entry_for_id(eid))
       .filter_map(move |entry| entry.peek(&context).map(|i| (entry.node(), i)))
   }
+
+  fn prune_unreacheable_from_relevant(&mut self) {
+    let dur = Duration::new(60, 0);
+    let now = Instant::now();
+    let relevant_root_ids: HashSet<_> = self
+      .roots_by_age
+      .iter()
+      .filter_map(|(k, v)| {
+        if now - *v < dur {
+          self.entry_id(k)
+        } else {
+          None
+        }
+      })
+      .collect();
+    let reachable_from_relevant: HashSet<_> = self
+      .walk(
+        relevant_root_ids.into_iter().cloned().collect(),
+        Direction::Incoming,
```

Comment:
This should be an `Outgoing` walk, which is a walk on dependencies: `Incoming` is a walk on dependents.